lib/LanguageServerRename/Adapter/Worse/MemberRenamer.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerRename\Adapter\Worse;
+
+use Generator;
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ConstElement;
+use Microsoft\PhpParser\Node\Expression\MemberAccessExpression;
+use Microsoft\PhpParser\Node\Expression\ScopedPropertyAccessExpression;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Microsoft\PhpParser\Node\MethodDeclaration;
+use Microsoft\PhpParser\Node\PropertyDeclaration;
+use Microsoft\PhpParser\Parser;
+use Microsoft\PhpParser\Token;
+use Phpactor\Extension\LanguageServerRename\Model\LocatedTextEdit;
+use Phpactor\Extension\LanguageServerRename\Model\Renamer;
+use Phpactor\ReferenceFinder\ReferenceFinder;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\ByteOffsetRange;
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\TextDocument\TextDocumentLocator;
+use Phpactor\TextDocument\TextEdit as PhpactorTextEdit;
+
+class MemberRenamer implements Renamer
+{
+    /**
+     * @var ReferenceFinder
+     */
+    private $referenceFinder;
+
+    /**
+     * @var TextDocumentLocator
+     */
+    private $locator;
+
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    public function __construct(
+        ReferenceFinder $referenceFinder,
+        TextDocumentLocator $locator,
+        Parser $parser
+    ) {
+        $this->referenceFinder = $referenceFinder;
+        $this->locator = $locator;
+        $this->parser = $parser;
+    }
+
+    public function getRenameRange(TextDocument $textDocument, ByteOffset $offset): ?ByteOffsetRange
+    {
+        $node = $this->parser->parseSourceFile($textDocument->__toString())->getDescendantNodeAtPosition($offset->toInt());
```

Comment:
The method is called for each reference, we can be parsing the same file multiple times. I think we still need the "group by document" logic.

Reply:
The calling class is responsible for grouping now, so `LocatedTextEditMap` collects all the location/edit objects and groups them by document.

Reply:
hm, sorry I see your point now. The parser is cached, so it shouldn't affect performance, but there's a better way,

Reply:
Saying that, the cache should be sufficient,

Reply:
It has cache? I didn't know that. Probably there is no reason to group them here now.